engine/dlib/src/dlib/connection_pool.cpp

```diff
@@ -249,80 +251,81 @@ namespace dmConnectionPool
         return false;
     }
 
-    static Result ConnectSocket(HPool pool, dmSocket::Address address, uint16_t port, int timeout, Connection* c, dmSocket::Result* sr)
+    static Result ConnectSocket(HPool pool, dmSocket::Address address, uint16_t port, int timeout, dmSocket::Socket* socket, dmSocket::Result* sr)
     {
-        *sr = dmSocket::New(address.m_family, dmSocket::TYPE_STREAM, dmSocket::PROTOCOL_TCP, &c->m_Socket);
+        *sr = dmSocket::New(address.m_family, dmSocket::TYPE_STREAM, dmSocket::PROTOCOL_TCP, socket);
         if (*sr != dmSocket::RESULT_OK) {
             return RESULT_SOCKET_ERROR;
         }
 
         if( timeout > 0 )
         {
-            *sr = dmSocket::SetBlocking(c->m_Socket, false);
+            *sr = dmSocket::SetBlocking(*socket, false);
             if (*sr != dmSocket::RESULT_OK) {
-                dmSocket::Delete(c->m_Socket);
+                dmSocket::Delete(*socket);
                 return RESULT_SOCKET_ERROR;
             }
 
-            *sr = dmSocket::Connect(c->m_Socket, address, port);
+            *sr = dmSocket::Connect(*socket, address, port);
             if (*sr != dmSocket::RESULT_OK) {
-                dmSocket::Delete(c->m_Socket);
+                dmSocket::Delete(*socket);
                 return RESULT_SOCKET_ERROR;
             }
 
             dmSocket::Selector selector;
             dmSocket::SelectorZero(&selector);
-            dmSocket::SelectorSet(&selector, dmSocket::SELECTOR_KIND_WRITE, c->m_Socket);
+            dmSocket::SelectorSet(&selector, dmSocket::SELECTOR_KIND_WRITE, *socket);
 
             *sr = dmSocket::Select(&selector, timeout);
             if( *sr == dmSocket::RESULT_WOULDBLOCK )
             {
-                dmSocket::Delete(c->m_Socket);
+                dmSocket::Delete(*socket);
                 return RESULT_SOCKET_ERROR;
             }
 
-            *sr = dmSocket::SetBlocking(c->m_Socket, true);
+            *sr = dmSocket::SetBlocking(*socket, true);
             if (*sr != dmSocket::RESULT_OK) {
-                dmSocket::Delete(c->m_Socket);
+                dmSocket::Delete(*socket);
                 return RESULT_SOCKET_ERROR;
             }
         }
         else
         {
-            *sr = dmSocket::Connect(c->m_Socket, address, port);
+            *sr = dmSocket::Connect(*socket, address, port);
             if (*sr != dmSocket::RESULT_OK) {
-                dmSocket::Delete(c->m_Socket);
+                dmSocket::Delete(*socket);
                 return RESULT_SOCKET_ERROR;
             }
         }
 
         return RESULT_OK;
     }
 
-    static Result Connect(HPool pool, const char* host, dmSocket::Address address, uint16_t port, bool ssl, int timeout, Connection* c, dmSocket::Result* sr)
+    static Result Connect(HPool pool, const char* host, dmSocket::Address address, uint16_t port, bool ssl, int timeout,
```

Comment:
Same here, no real need for the `Connection* c`, so now it's focusing on the actuual sockets.